Controllers/ReferenceController.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using ReferenceReferralApp.Models;
+
+namespace ReferenceReferralApp.Controllers
+{
+    public class ReferenceController : Controller
+    {
+        private readonly ReferenceContext _context;
+
+        public ReferenceController(ReferenceContext context)
+        {
+            _context = context;
+        }
+
+        // GET: Reference
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.References.ToListAsync());
+        }
+
+        // GET: Reference/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var reference = await _context.References
```

Comment:
If you think about SRP (single responsbility principle) every class and function should do one job, and only one job well. In this case, I would move your queries to a respository class. Call that repository class from your controllers. That provides a few benefits 1. The code is more maintainable through SRP 2. The queries now become testable if you so choose to do that 3. It provides the ability to reuse those queries in other classes or controllers.